### PR TITLE
Updates DT Calibration to 14_0_X and includes fixes for the HI workflow

### PR DESCRIPTION
**Pull request by ftorrresd**
#### PR description: 

Following the discussion from [1], this PR updates the DT Calibration code to CMSSW_14_0_X and includes fixes for the HI workflow (available trigger paths + "rawDataCollector" --> "rawDataRepacker").

[1] - https://indico.cern.ch/event/1339792/#80-dt-calibration-results-with

##### pp - Run2023

```bash
dtCalibration \
       ttrig \
       residuals \
       all \
       --run=370175 \
       --trial=3 \
       --label=PPtest \
       --runselection=370175 \
       --datasetpath=/ExpressPhysics/Run2023D-Express-v1/FEVT \
       --globaltag=130X_dataRun3_HLT_v2 \
       --lumisPerJob 50 \
       --preselection=CalibMuon.DTCalibration.dtCalibOfflineSelection_cff:dtCalibOfflineSelection \
       --output-site T2_CH_CERN \
       --user YOUR_CERN_USERNAME \
       --run-on-RAW \
       --max-checks 100000
```

##### HI - Run2023

```bash
dtCalibration \ 
       ttrig \
       residuals\
       all\
       --run=375831\
       --trial=6\
       --label=HItest\
       --runselection=375823\
       --datasetpath=/HIExpressPhysics/HIRun2023A-Express-v2/FEVT\
       --globaltag=132X_dataRun3_Express_v4\
       --lumisPerJob 50\
       --preselection=CalibMuon.DTCalibration.dtCalibOfflineSelection_cff:dtCalibOfflineSelectionHI
       --raw-data-label=rawDataRepacker
       --output-site T2_CH_CERN
       --user YOUR_CERN_USERNAME
       --run-on-RAW
       --max-checks 100000
       --ce-white-list=T2_FR_IPHC
```

##### Cosmics - Run2023

```bash
dtCalibration \
        ttrig \
        timeboxes \
        all \
        --run=366296  \
        --trial=3 \
        --label=Cosmicstest \
        --runselection=366296 \
        --datasetpath=/ExpressCosmics/Run2023A-Express-v2/FEVT \
        --globaltag=130X_dataRun3_Express_v2 \
        --lumisPerJob 50 \
        --preselection=CalibMuon.DTCalibration.dtCalibOfflineSelection_cff:dtCalibOfflineSelectionCosmics \
        --output-site T2_CH_CERN \
        --user YOUR_CERN_USERNAME \
        --run-on-RAW \
       --max-checks 100000
```

No significant difference was observed.

@battibass @fcavallo @valesark

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
@ossobooker where do the validations that you list in the PR description should be run?
I'm trying to run those commands in lxplus in some recent IB + this PR, but they do not seem to work.
Please provide a working recipe.

File: CalibMuon/DTCalibration/python/dtCalibOfflineSelection_cff.py [hunk 1]
```diff
@@ -16,8 +16,9 @@
 hltDTCalibration.HLTPaths = ['HLT_DTCalibration_v*']
 
 ALCARECODtCalibHIHLTFilter = copy.deepcopy(hltHighLevel)
-ALCARECODtCalibHIHLTFilter.throw = False
-ALCARECODtCalibHIHLTFilter.eventSetupPathsKey = 'MuAlcaDtCalibHI'
+
+ALCARECODtCalibHIHLTFilter.HLTPaths = ['HLT_HIL2SingleMu*']
+
```
Comment (perrotta) [inline CalibMuon/DTCalibration/python/dtCalibOfflineSelection_cff.py:21]:
```suggestion
ALCARECODtCalibHIHLTFilter.HLTPaths = ['HLT_HIL2SingleMu*']
```
(not worth updating the PR only for this, but if you need to for a different reason please take it into account)

**Comment by valesark** [comment]
> @ossobooker where do the validations that you list in the PR description should be run? I'm trying to run those commands in lxplus in some recent IB + this PR, but they do not seem to work. Please provide a working recipe.

Dear @perrotta, in order to be able to run the commands mentioned in the PR description, first of all, you should have activated `cmsenv` and compiled the project `scram b -j4`. Once this is done, you should go to the `CalibMuon/DTCalibration` subdirectory and the aforementioned commands should work from there. Please make sure that your proxy (`voms-proxy-init --voms cms --valid 168:00`) is activated as well. Let me know if you need further help with this.

**Comment by perrotta** [comment]
+1
- Tests succeed (thank you @battibass for the support)

**Comment by rappoccio** [comment]
+1

**Pull request merged**